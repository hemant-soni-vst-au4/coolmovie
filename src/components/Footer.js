import React from "react";

class Footer extends React.Component {
  render() {
    return (
      <div style={{ textAlign: "center", marginTop: 20 }}>
        <p>
          Created by Hemant Soni. Powered by{" "}
          <img
            style={{ height: 50 }}
            src={require("../assets/img/moviedblogo.svg")}
            alt="moviedblogo"
          />
        </p>
      </div>
    );
  }
}

export default Footer;