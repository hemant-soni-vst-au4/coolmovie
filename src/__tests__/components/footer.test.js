import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import {
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from '../../components/Footer';

describe('Render the page correctly', () => {
  test('Render correctly', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Footer element image in place', () => {
    const { container } = render(<Footer />);
    expect(container.innerHTML).toContain('<img style="height: 50px;" src="moviedblogo.svg" alt="moviedblogo">');
  });

  test('Footer content in place', () => {
    expect('Created by Hemant Soni. Powered by ').toMatchSnapshot();
  });
});
