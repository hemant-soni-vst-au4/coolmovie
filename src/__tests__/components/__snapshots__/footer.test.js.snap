// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the page correctly Footer content in place 1`] = `"Created by Hemant Soni. Powered by "`;

exports[`Render the page correctly Render correctly 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <p>
      Created by Hemant Soni. Powered by 
      <img
        alt="moviedblogo"
        src="moviedblogo.svg"
        style="height: 50px;"
      />
    </p>
  </div>
</DocumentFragment>
`;
